packages/common/src/authorization/organizationMemberAbility.ts

```diff
@@ -0,0 +1,71 @@
+import { Ability, AbilityBuilder, ForcedSubject } from '@casl/ability';
+import { Organization } from '../types/organization';
+import {
+    OrganizationMemberProfile,
+    OrganizationMemberRole,
+} from '../types/organizationMemberProfile';
+
+type Action = 'manage' | 'update' | 'view' | 'create' | 'delete';
+
+type Subject =
+    | Organization
+    | OrganizationMemberProfile
+    | 'Organization'
+    | 'OrganizationMemberProfile'
+    | 'Dashboard'
+    | 'SavedChart'
+    | 'Project';
```

Comment:
What is the use case for using the types + strings instead of just strings ?

Reply:
In CASL you can validate against a subject type or a subject instance: https://casl.js.org/v5/en/guide/intro#checking-logic

So a user ability might have `can('update', 'Organisation') => true` meaning that the user can manage **at least one organization**.

But then if they check `can('update', subject('Organisation', { organizationUuid: '1234' }) => false` so they can't access that specific type. Adding both in the `type` helps with type completion. 

Separately: there is actually a bug where the `subject(...)` doesn't show the narrow types (it's just any).